src/backend/libc/process/syscalls.rs

```diff
@@ -646,3 +655,27 @@ pub(crate) fn getgroups(buf: &mut [Gid]) -> io::Result<usize> {
 
     unsafe { ret_usize(c::getgroups(len, buf.as_mut_ptr().cast()) as isize) }
 }
+
+#[cfg(not(any(
+    target_os = "emscripten",
+    target_os = "espidf",
+    target_os = "fuchsia",
+    target_os = "redox",
+    target_os = "vita",
+    target_os = "wasi"
+)))]
+#[inline]
+pub(crate) fn fcntl_getlk(fd: BorrowedFd<'_>, lock: &Flock) -> io::Result<Option<Flock>> {
+    if lock.typ == crate::process::FlockType::Unlocked {
+        return Err(io::Errno::INVAL);
+    }
+
+    let mut curr_lock: c::flock = lock.as_raw();
+    unsafe { ret(c::fcntl(borrowed_fd(fd), c::F_GETLK, &mut curr_lock))? };
+
+    if curr_lock.l_pid == Pid::as_raw(lock.pid) {
```

Comment:
Could you comment on what this code is doing? I'm looking at the documentation for `F_GETLK` and it's not clear what this condition corresponds to.